### PR TITLE
ci:  Add pytorch2 tests

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Description

Putting our pytorch single-gpu and cpu tests on the shiny new `pytorch==2.0.1` images. Also throwing one nightly test in for good measure.

## Test Plan

`test-unit-harness-pytorch2-cpu`
`test-unit-harness-pytorch2-gpu`

## Commentary (optional)

Hopefully we will not need this in the future when we upgrade our default task environments. We'll treat these as separate for now, until we have a better way to verify that they work with every integration we support.

## Checklist

- [ ] Changes have been manually QA'd
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
MLG-510

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1869,6 +1869,52 @@ jobs:
...
+      - skip-if-github-only
+      - skip-if-webui-only
+      - run: pip install mypy pytest coverage
+      - install-codecov
+      - run: echo 'export PATH=$PATH:$HOME/.local/bin' >> $BASH_ENV
+      - run: echo 'export PATH=$PATH:/usr/local/nvidia/bin' >> $BASH_ENV
+      - run: echo 'export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/local/nvidia/lib64' >> $BASH_ENV
+      - run: COVERAGE_FILE=/root/project/test-unit-harness-pytorch2-cpu-pycov make -C harness test-pytorch-cpu
+      - run: coverage xml -i --data-file=./test-unit-harness-pytorch2-cpu-pycov
+      - run: codecov -v -t $CODECOV_TOKEN -F harness
```
Comment (ioga) [inline .circleci/config.yml:1911]: can we not repeat these somewhat arcane steps every time, and instead have a `test-unit-harness` job parametetrized by
- docker image
- test name
- persisted coverage file path

File: docs/release-notes/pytorch2.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+:orphan:
+
+**New Features**
+
+-  Environments: Add PyTorch2.0 Beta images.
+
+   -  We are providing a set of images with `pytorch==2.0.1`, `python==3.10.12`, and `CUDA==11.8`.
+      These images have not been fully tested against certain integrations, and are still in beta.
```
Comment (ioga) [inline docs/release-notes/pytorch2.rst:8]: this is a docs question but we have never used the notion of `beta` before. these things has always been branded as "experimental".

File: e2e_tests/tests/config.py [hunk 3]
```diff
@@ -176,6 +180,10 @@ def set_pt_image(config: Dict[Any, Any]) -> Dict[Any, Any]:
     return set_image(config, PT_CPU_IMAGE, PT_GPU_IMAGE)
 
 
+def set_pt2_image(config: Dict[Any, Any]) -> Dict[Any, Any]:
```
Comment (ioga) [inline e2e_tests/tests/config.py:183]: don't see this method being called anywhere?

**Review by mpkouznetsov — APPROVED** [review]
looks good.

File: e2e_tests/tests/config.py (hunk 3 above)
Comment (MikhailKardash) [reply e2e_tests/tests/config.py:183]: Oops, nice catch. Meant to add it to a nightly test.

File: docs/release-notes/pytorch2.rst (hunk 2 above)
Comment (MikhailKardash) [reply docs/release-notes/pytorch2.rst]: I updated the phrasing, but I'll also wait for docs to review.

File: .circleci/config.yml (hunk 1 above)
Comment (MikhailKardash) [reply .circleci/config.yml]: I condensed these a little bit just for good measure here, but not every one of these jobs is the same. Some cpu jobs use docker, some use the old venv structure, GPU jobs have an extra library update step. A redesign is probably in order, but I'd make that a separate PR.

File: docs/release-notes/pytorch2.rst [hunk 4]
```diff
@@ -0,0 +1,8 @@
+:orphan:
+
+**New Features**
+
+-  Environments: Add experimental PyTorch2.0 images.
+
+   -  We are providing a set of images with `pytorch==2.0.1`, `python==3.10.12`, and `CUDA==11.8`.
+      These images have not been fully tested against certain integrations, and are experimental.
```
Comment (tara-det-ai) [inline docs/release-notes/pytorch2.rst:8]: :orphan:

**New Features**

-  Environments: Add the following PyTorch2.0 images (experimental): ``pytorch==2.0.1``, ``python==3.10.12``, and ``CUDA==11.8``.

Comment (tara-det-ai) [reply docs/release-notes/pytorch2.rst:8]: @dzhu please review my proposed edit

**Review by dzhu — APPROVED** [review]
Looks reasonable, though it looks like you should be able to deduplicate the entirety of these jobs, rather than stopping with setting up the paths. But that doesn't need to happen here.

**Comment by dzhu** [comment]
Also reminder to, before landing the PR, delete the many unnecessary lines in the commit message (one for each pre-land commit) that GitHub will insert by default. And write something useful in that space if you're so inclined.

**Pull request merged**